internal/ingress/controller/parser/parser.go

```diff
@@ -1740,26 +1773,55 @@ func toKongPlugin(plugin plugin) kong.Plugin {
 	return result
 }
 
-func kongPluginFromK8SClusterPlugin(k8sPlugin configurationv1.KongClusterPlugin) kong.Plugin {
-	return toKongPlugin(plugin{
+func (p *Parser) kongPluginFromK8SClusterPlugin(k8sPlugin configurationv1.KongClusterPlugin) (kong.Plugin, error) {
+	var err error
+	var configError error
+	config := k8sPlugin.Config
+	if k8sPlugin.ConfigFrom != (configurationv1.NamespacedSecretValueFromSource{}) {
+		if len(k8sPlugin.Config) > 0 {
+			return kong.Plugin{}, errors.Errorf("cluster plugin '%v' has both Config and ConfigFrom set", k8sPlugin.Name)
+		}
+		config, configError = p.namespacedSecretToConfiguration(k8sPlugin.ConfigFrom)
+		if configError != nil {
+			err = fmt.Errorf("error parsing config for cluster plugin %v: %w", k8sPlugin.Name, configError)
+		}
+		k8sPlugin.Config = config
+	}
```

Comment:
We can and should improve the readability of this `if` block.
Adopt a simpler approach like so:
- check early, if config and configFrom are both set, return err
- if config is set, then set that
- if configFrom is set, then set that, any error in this and the previous step, return err
- construct the result and return (w/o error)

Reply:
Not sure if I follow entirely. ac1429a9776b950050d03c171f88ee36a851d391 moves the conflict check into its own block, and removes the now superfluous `err` (we only returned it on success), but I'm not sure what we can do with the rest. `config` is already set: it's being stored temporarily so we can return it later if there's no `configFrom` or overwrite if there is, without modifying the `k8splugin` argument.